packages/bpk-dialog-utils/src/BpkDialogWrapper.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { ReactNode } from "react";
+import { useEffect, useRef } from "react";
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import CSSTransition from 'react-transition-group/CSSTransition';
+
+import { cssModules } from '../../bpk-react-utils';
+
+import STYLES from './BpkDialogWrapper.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export interface Props {
+  ariaLabelledby: string;
+  children: ReactNode;
+  closeOnEscPressed?: boolean;
+  closeOnScrimClick?: boolean;
+  dialogClassName?: string;
+  id: string | undefined;
+  isOpen: boolean;
+  onClose: (
+    arg0?: Event | KeyboardEvent | MouseEvent | PointerEvent | TouchEvent,
+    arg1?: {
+      source: 'ESCAPE' | 'DOCUMENT_CLICK';
+    },
+) => void | null;
+  exiting?: boolean;
+  transitionClassNames?: {
+    appear?: string,
+    appearActive?: string,
+    exit?: string
+  };
+  timeout?: {appear?: number, exit?: number};
+};
+
+type DialogProps = {
+  isDialogOpen: boolean;
+};
+
+// TODO: this check if the browser support the HTML dialog element. We can remove it once we drop support as a business for Safari 14
+const dialogSupported = typeof HTMLDialogElement === 'function';
```

Comment:
check if dialog is supported or not; and then use the polyfill if not supported 💪